src/Protector.php

```diff
@@ -296,7 +293,7 @@ public function getRemoteDump(): string
             ($destinationFilename ?? 'remote_dump.sql')
         );
 
-        $disk->put($destinationFilepath, $body);
+        $disk->put($destinationFilepath, $this->decryptDump($body));
```

Comment:
This might be an issue with big database dumps. You load the encrypted body inside the memory and then have sodium decrypt it in memory, which might double the memory consumption... can't we stream it instead?

Reply:
this is exactly as it was before. a streaming solution is already in the works i hear tho.

Reply:
will be done in another branch.